omp/matrix/hybrid_kernels.cpp

```diff
@@ -100,7 +101,88 @@ template <typename ValueType, typename IndexType>
 void convert_to_csr(std::shared_ptr<const OmpExecutor> exec,
                     matrix::Csr<ValueType, IndexType> *result,
                     const matrix::Hybrid<ValueType, IndexType> *source)
-    GKO_NOT_IMPLEMENTED;
+{
+    auto csr_val = result->get_values();
+    auto csr_col_idxs = result->get_col_idxs();
+    auto csr_row_ptrs = result->get_row_ptrs();
+    const auto ell = source->get_ell();
+    const auto max_nnz_per_row = ell->get_num_stored_elements_per_row();
+    const auto coo_val = source->get_const_coo_values();
+    const auto coo_col = source->get_const_coo_col_idxs();
+    const auto coo_row = source->get_const_coo_row_idxs();
+    const auto coo_nnz = source->get_coo_num_stored_elements();
+    const auto num_rows = source->get_size()[0];
+    auto coo_row_ptrs = Array<IndexType>(exec, num_rows + 1);
+    auto coo_row_ptrs_val = coo_row_ptrs.get_data();
+    convert_sorted_idxs_to_ptrs(coo_row, coo_nnz, coo_row_ptrs_val,
+                                num_rows + 1);
+
+    // Compute the row offset of Coo without zeros
+    auto coo_offset = Array<IndexType>(exec, num_rows);
+    auto coo_offset_val = coo_offset.get_data();
+    for (size_type i = 0; i < num_rows; i++) {
+        IndexType nonzeros = 0;
+#pragma parallel for reduction(+ : nonzeros)
+        for (size_type j = coo_row_ptrs_val[i]; j < coo_row_ptrs_val[i + 1];
+             j++) {
+            nonzeros += coo_val[j] != zero<ValueType>();
+        }
+        coo_offset_val[i] = nonzeros;
+    }
+
+    // Compute row pointer of Csr
+    csr_row_ptrs[0] = 0;
+#pragma parllel for
+    for (size_type i = 0; i < num_rows; i++) {
+        csr_row_ptrs[i + 1] = coo_offset_val[i];
+    }
+
+    for (size_type col = 0; col < max_nnz_per_row; col++) {
+#pragma parallel for
+        for (size_type row = 0; row < num_rows; row++) {
+            csr_row_ptrs[row + 1] +=
+                (ell->val_at(row, col) != zero<ValueType>());
+        }
+    }
+
+    auto workspace = Array<IndexType>(exec, num_rows + 1);
+    auto workspace_val = workspace.get_data();
+    for (size_type i = 1; i < num_rows + 1; i <<= 1) {
+#pragma parallel for
+        for (size_type j = i; j < num_rows + 1; j++) {
+            workspace_val[j] = csr_row_ptrs[j] + csr_row_ptrs[j - i];
+        }
+#pragma parallel for
```

Comment:
Same here for eventual slowdown, also some internal reductions in some previous code.